src/net/sourceforge/kolmafia/webui/FightDecorator.java

```diff
@@ -271,6 +276,37 @@ private static void decorateSourceAgent(final StringBuffer buffer) {
     }
   }
 
+  // <form name=useitem action=fight.php method=post style='display: inline; margin: 0 0.5em 0
+  // 0.5em'><input type=hidden name=whichskill value=7512d><input type=hidden name=action
+  // value="skill"><input class=button type=submit onclick="return killforms(this);" value="Advanced
+  // Research"></form>
+
+  // <form name=useitem action=fight.php method=post style='color=graydisplay: inline; margin: 0
+  // 0.5em 0 0.5em'><input type=hidden name=whichskill value=7512d><input type=hidden name=action
+  // value="skill"><input class=button type=submit onclick="return killforms(this);" value="Advanced
+  // Research"></form>
+
+  private static final Pattern ADVANCED_RESEARCH_BUTTON =
+      Pattern.compile("<form name=useitem.*?value=\"Advanced Research\"></form>", Pattern.DOTALL);
+
+  private static void decorateAdvancedResearch(
+      final MonsterData monster, final StringBuffer buffer) {
+    // Extract the "skill" form from the buffer
+    Matcher matcher = ADVANCED_RESEARCH_BUTTON.matcher(buffer);
+    if (!matcher.find()) {
+      return;
+    }
+    // If we have previously researched this monster, disable the Advanced Research button
+    if (!FightRequest.hasResearchedMonster(monster.getId())) {
+      return;
+    }
+    String form = matcher.group(0);
+    String replacement =
+        StringUtilities.singleStringReplace(
+            form, "class=button", " disabled style='color=gray' class=button");
```

Comment:
style takes CSS, so I think this should be `style='color:gray'`?

Reply:
That is correct. Thanks. However, I have a better solution. I'm going to simply delete the button, just like KoL does in subsequent rounds after you've done Advanced Research. Why bother showing it - even disabled - if using it would simply give an error message.